### PR TITLE
Rego format

### PR DESCRIPTION
**Pull request by wcalderipe**
This PR adds a base rule in the `.editorconfig` to format Rego files using the built-in formatter in the OPA CLI. It also adds a step in the workflow to check the format to ensure consistency.

Reference https://docs.styra.com/opa/rego-style-guide#use-opa-fmt

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -14,3 +14,11 @@ trim_trailing_whitespace = false
...
 [Makefile]
 indent_style = tab
+
+# See https://docs.styra.com/opa/rego-style-guide
+[*.rego]
+end_of_line = lf
+insert_final_newline = true
+charset = utf-8
+indent_style = tab
+indent_size = 4
```
Comment (wcalderipe) [inline .editorconfig:24]: This is important.

File: .github/workflows/policy-engine.yml [hunk 2]
```diff
@@ -128,7 +128,10 @@ jobs:
         with:
           version: latest
 
-      - name: Test rego
+      - name: Code format
+        run: make policy-engine/rego/format/check
+
+      - name: Test
```
Comment (wcalderipe) [inline .github/workflows/policy-engine.yml:134]: This is important.

File: apps/policy-engine/Makefile [hunk 3]
```diff
@@ -34,18 +34,18 @@ policy-engine/build:
```
Comment (wcalderipe) [inline apps/policy-engine/Makefile:34]: These changes are important.

File: apps/policy-engine/src/resource/open-policy-agent/rego/__test__/criteria/intent/contractCall_test.rego [hunk 4]
```diff
@@ -10,7 +10,7 @@ test_contractCall {
 			"type": "contractCall",
```
Comment (wcalderipe) [inline apps/policy-engine/src/resource/open-policy-agent/rego/__test__/criteria/intent/contractCall_test.rego:10]: Down this file, is just formatted code.

**Review by samteb — APPROVED** [review]

**Pull request merged**